tests/vstools/utils/test_props.py

```diff
@@ -0,0 +1,178 @@
+
+        with self.assertRaisesRegex(FramePropError, func_name):
+            get_prop(clip, "NonExistentProp", int, func=func_name)
+
+    def test_get_prop_cast_int(self) -> None:
+        """Test get_prop casting to int."""
+
+        self.assertEqual(get_prop(clip, "__IntProp", int, cast=int), 123)
+        self.assertEqual(get_prop(clip, "__FloatProp", float, cast=int), 123)
+        self.assertEqual(get_prop(clip, "__BoolProp", int, cast=int), 1)
+
+        self.assertRaises(FramePropError, get_prop, clip, "__StrProp", str, cast=int)
+        self.assertRaises(FramePropError, get_prop, clip, "__BytesProp", bytes, cast=int)
+        self.assertRaises(FramePropError, get_prop, clip, "__VideoFrameProp", vs.VideoFrame, cast=int)
+
+    def test_get_prop_cast_float(self) -> None:
+        """Test get_prop casting to float."""
+
+        self.assertEqual(get_prop(clip, "__IntProp", int, cast=float), 123.0)
+        self.assertEqual(get_prop(clip, "__FloatProp", float, cast=float), 123.456)
+        self.assertEqual(get_prop(clip, "__BoolProp", int, cast=float), 1.0)
+
+        self.assertRaises(FramePropError, get_prop, clip, "__StrProp", str, cast=float)
+        self.assertRaises(FramePropError, get_prop, clip, "__BytesProp", bytes, cast=float)
+        self.assertRaises(FramePropError, get_prop, clip, "__VideoFrameProp", vs.VideoFrame, cast=float)
+
+    def test_get_prop_cast_bool(self) -> None:
+        """Test get_prop casting to bool."""
+
+        self.assertEqual(get_prop(clip, "__StrProp", str, cast=bool), True)
+        self.assertEqual(get_prop(clip, "__IntProp", int, cast=bool), True)
+        self.assertEqual(get_prop(clip, "__FloatProp", float, cast=bool), True)
+        self.assertEqual(get_prop(clip, "__BoolProp", int, cast=bool), True)
+        self.assertEqual(get_prop(clip, "__BytesProp", bytes, cast=bool), True)
+        self.assertEqual(get_prop(clip, "__VideoFrameProp", vs.VideoFrame, cast=bool), True)
+
+    def test_get_prop_cast_str(self) -> None:
+        """Test get_prop casting to str."""
+
+        self.assertEqual(get_prop(clip, "__StrProp", str, cast=str), "test string")
+        self.assertEqual(get_prop(clip, "__IntProp", int, cast=str), "123")
+        self.assertEqual(get_prop(clip, "__FloatProp", float, cast=str), "123.456")
+        self.assertEqual(get_prop(clip, "__BoolProp", int, cast=str), "1")
+        self.assertEqual(get_prop(clip, "__BytesProp", bytes, cast=str), b"test bytes")
+        self.assertEqual(get_prop(clip, "__VideoFrameProp", vs.VideoFrame, cast=str), str(clip.get_frame(0)))
+
+    def test_get_prop_cast_bytes(self) -> None:
+        """Test get_prop casting to bytes."""
+
+        self.assertEqual(get_prop(clip, "__StrProp", str, cast=bytes), b"test string")
+        self.assertEqual(get_prop(clip, "__BytesProp", bytes, cast=bytes), b"test bytes")
+        self.assertEqual(get_prop(clip, "__VideoFrameProp", vs.VideoFrame, cast=bytes), bytes(clip.get_frame(0)))
```

Comment:
`bytes(clip.get_frame(0))` throws an error `TypeError: 'memoryview' object cannot be interpreted as an integer` when just calling that in a test file.